api.js

```diff
@@ -1,50 +1,58 @@
 const express = require('express');
 const app = express();
 const path = require('path');
-const ytdl = require('ytdl-core');
-const { spawn } = require('child_process')
 
-let playing = false;
+const AudioStream = require('./AudioStream')
+let audioStream = new AudioStream();
 
 // Serve static files from the 'public' directory
 app.use(express.static(path.join(__dirname, 'public')));
 
 // Endpoint to fetch video info from YouTube URL
-app.get('/stream', async (req, res) => {
-    try {
-        const videoUrl = req.query.url;
-        streamYouTubeAudio(videoUrl);
-        
-        res.json(playing);
-    } catch (error) {
-        res.status(500).json({ error: 'An error occurred while fetching video info.' });
-    }
+app.post('/stream', async (req, res) => {
+  try {
+    const videoUrl = req.query.url;
+    audioStream.streamYouTubeAudio(videoUrl);
+    
+    res.json({ playing: audioStream.isPlaying() });
+  } catch (error) {
+    res.status(500).json({ error: 'An error occurred while fetching video info.' });
```

Comment:
What do you think of adding some dynamic information so we can see if the URL was correct or if it was received properly.
```suggestion
    res.status(500).json({ error: `An error occurred while fetching ${req.query.url}` });
```